elastica/memory_block/memory_block_rod.py

```diff
@@ -97,6 +92,67 @@ def __init__(self, systems: Sequence):
             (self.ghost_voronoi_idx[::3], self.n_voronoi)
         )
 
+        # Periodic boundaries are only used if there is a ring rod present in the simulation, otherwise below
+        # arrays are empty.
+        (
+            _,
+            self.periodic_boundary_nodes_idx,
+            self.periodic_boundary_elems_idx,
+            self.periodic_boundary_voronoi_idx,
+        ) = make_block_memory_periodic_boundary_metadata(n_elems_ring_rods)
+
+        """
+        If there are ring rods appended to simulation, then start and end idx of rod nodes, elements and voronoi
+        have to be updated, to accommodate the periodic boundaries. Periodic boundaries can only accessed by
+        memory block  and rods does not have access to periodic boundaries.
```

Comment:
This is a bit confusing to me, I would intuit that the concept of periodicity lies on the rod, while the block is just a computational and memory unit that crunches numbers. Maybe this structure should be revisited later.

Reply:
This can be revisited in future. But idea is we need to introduce periodicity, and I did not want user to be exposed to periodic elements, nodes, or voronoi. So I did this implementation in block structure.